search/search-dev-tools/src/components/query/index.js

```diff
@@ -0,0 +1,153 @@
+import 'prism-themes/themes/prism-ghcolors.css';
+import Editor from 'react-simple-code-editor';
+import cx from 'classnames';
+import pluralize from 'pluralize';
+
+
+import { SearchContext } from '../../context';
+import { postData } from '../../utils';
+import { CollapsibleList } from '../collapsible-list';
+
+import style from './style.scss';
+
+/**
+ * A single query
+ *
+ * @returns {preact.VNode} A query component.
+ */
+const Query = ( { args, request, url, query_args, backtrace = [] } ) => {
+	const txtQuery = JSON.stringify( args.body, null, 2 );
+	const txtResult = JSON.stringify( request.body, null, 2 );
+	const initialState = {
+		editing: false,
+		query: txtQuery,
+		result: txtResult,
+		collapsed: true,
+	};
+
+	const [ state, setState ] = useState( initialState );
+
+	const queryResultRef = useRef( null );
+
+	/**
+	 * 
+	 * @param {Object} query the query to Run
+	 * @param {String} url ES url
+	 */
+	const fetchForQuery = async ( query, url ) => {
+		try {
+			const res = await postData( window.VIPSearchDevTools.ajaxurl, {
+				action: window.VIPSearchDevTools.action,
+				url,
+				query,
+			}, window.VIPSearchDevTools.nonce );
+
+			setState({ ...state, result: JSON.stringify( res?.result?.body, null, 2 ) } );
+		} catch ( err ) {
+			// eslint-disable-next-line no-console
+			console.log( err );
+		}
+	};
+
+	useEffect( () => {
+		// Skip remote fetching if the query is the same.
+		// (e.g. was just reset to the initial one or wasn't change at all)
+		if ( state.query === initialState.query ) {
+			return;
+		}
+
+		if ( ! state.editing ) {
+			fetchForQuery( state.query, url );
+		}
+	}, [ state.query, state.editing ] );
+
+	// Re-highlight the query on result change (after the response is received).
+	useEffect( () => {
+		highlightElement( queryResultRef.current );
+	}, [ queryResultRef, state.result ] );
+
+	return ( <div className={cx( style.query_wrap, state.collapsed ? style.query_collapsed : null )}>
+		<div className={style.query_handle} onClick={ () => setState( { ...state, collapsed: ! state.collapsed } ) }>
+			<h3 className="vip-h3">
+				{ pluralize( 'result', ( request?.body?.hits?.hits?.length || 0 ), true )}
+				<span style="color: var(--vip-grey-60);"> that took</span> {request.body.took}ms
+				<small> ({request.response.code})</small>
+			</h3>
+		</div>
+		<div className={style.grid_container}>
+			<div className={style.query_src_header}>
+				<span style="margin-right: auto;">Request</span>
+				<div className={style.query_src_extra}>
+					<CollapsibleList title="WP_Query"
+						list={ Object.entries( query_args ).map( ( [ key, value ] ) => `${ key }: ${ JSON.stringify( value ) }` ) }
+					/>
+					<CollapsibleList title="Trace" list={ backtrace } />
+				</div>
+			</div>
+			<div className={style.query_res_header}>
+				Response
+			</div>
+			<div className={style.query_src}>
+				<div className={style.query_actions}>
+					{ state.editing || state.result !== txtResult
+						? ( <Fragment>
```

Comment:
React has fragment shortcut -  `<>...</>`. Does that work in preact as well?

Reply:
@pschoffer It used to be that Preact only supported `<Fragment>` but I just checked the docs and they say it's supported now. Let me try!

Reply:
It works, but with a caveat that the named import should still use Fragment, removing the import blows the bundle size up 400b (gzipped).